webapp/channels/src/packages/mattermost-redux/src/actions/profiles_statuses_groups.ts

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+/* eslint-disable no-console */
+
+import type {PostList, Post} from '@mattermost/types/posts';
+
+import {getProfilesByIds, getStatusesByIds} from 'mattermost-redux/actions/users';
+import {getCurrentUserId, getIsUserStatusesConfigEnabled} from 'mattermost-redux/selectors/entities/common';
+import type {ActionFunc} from 'mattermost-redux/types/actions';
+
+const pendingUserIdsForProfiles = new Set<string>();
+const MAX_USER_PROFILES_BATCH = 100;
+const USER_PROFILES_DURATION = 10 * 1000;
+
+let userProfilesIntervalId: NodeJS.Timeout | null = null;
+
+const pendingUserIdsForStatuses = new Set<string>();
+const MAX_USER_STATUSES_BATCH = 100;
+const USER_STATUSES_DURATION = 20 * 1000;
+
+let userStatusesIntervalId: NodeJS.Timeout | null = null;
+
+function addUserIdForProfiles(userId: string): ActionFunc<boolean> {
+    return (dispatch) => {
+        function getPendingProfilesById() {
+            console.log('addUserIdForProfiles start', pendingUserIdsForProfiles);
+
+            dispatch(getProfilesByIds(Array.from(pendingUserIdsForProfiles)));
```

Comment:
Should we not run this through a dedup filter?

Reply:
This is using a Set, so it's automatically deduped by that